### ISSUE TITLE
oci-adb-client-runsql-python - Issue

### ISSUE THREAD

**Issue by tanker110** [oracle/oracle-functions-samples#37]
Get the following issue when trying to deploy  oci-adb-client-runsql-python sample:

 Complete output from command python setup.py egg_info:
    
            =============================DEBUG ASSISTANCE==========================
            If you are seeing an error here please try the following to
            successfully install cryptography:
    
            Upgrade to the latest pip and try again. This will fix errors for most
            users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
            =============================DEBUG ASSISTANCE==========================

**Comment by sachin-pikle** [comment]
Hi @tanker110, is this the error you are seeing "ModuleNotFoundError: No module named 'setuptools_rust'" in the last line of the output when you run "fn -v build" or while run the deploy command?

If yes, then can you set the version of the "oci" python sdk to 2.40.0 in function's the "requirements.txt" ... We have updated the function sample with this change.
oci==2.40.0

Looks like there is a new dependency introduced in oci python sdk library from version 2.40.1. The cryptography<=3.4.7,>=3.2.1 results in an error "ModuleNotFoundError: No module named 'setuptools_rust'"

Please reopen the ticket is you run into issues with the suggested change.

**Comment by sachin-pikle** [comment]
Hi @tanker110, we have updated the Dockerfile and func.yaml for this sample to use the latest OCI Python SDK (versions 2.40.1+). We had to update pip in the Dockerfile. The OCI Python SDK (version 2.40.1) updated the Cryptography versions. The new Cryptography uses a newer version of setuptools which is not available in Pip. Updating Pip gives you access to download that.

**Comment by tanker110** [comment]
Hi Sachin,

Thank you so much. Before I heard from you I was able to add pip upreade and install rust to the docker file. Thanks so much for your response.

We are doing a proof of concept of moving our infrastructure to the cloud and was testing some of the function ability. I too the python using client and changed it a bit
To use an event that would get trigerred when users drop files into object storage bucket. It would read the file name from the JSON event and pass it to oracle to run a procedure.

I’ve never done anything in python so it was unique experience. I did have a question about the DB ENCRYPT password. Is that added in plain text in the configuration parameters on the OCI interfave and then is encrypted once the call to the function is made? I wasn’t sure how that worked. I also added a try catch when I call the procedure so I could catch Object not found. For example
If a user dropped a file in the bucket then immediately removed it. I wanted to be able to log a clean Object not found in bucket. So far it seems to work.

Thanks,
Tommy
